extension/core_functions/scalar/list/array_slice.cpp

```diff
@@ -59,18 +59,25 @@ static idx_t CalculateSliceLength(idx_t begin, idx_t end, INDEX_TYPE step, bool
 }
 
 template <typename INPUT_TYPE, typename INDEX_TYPE>
-INDEX_TYPE ValueLength(const INPUT_TYPE &value) {
+INDEX_TYPE ValueLength(const INPUT_TYPE &value, const LogicalTypeId &) {
 	return 0;
 }
 
 template <>
-int64_t ValueLength(const list_entry_t &value) {
+int64_t ValueLength(const list_entry_t &value, const LogicalTypeId &) {
 	return UnsafeNumericCast<int64_t>(value.length);
 }
 
 template <>
-int64_t ValueLength(const string_t &value) {
-	return LengthFun::Length<string_t, int64_t>(value);
+int64_t ValueLength(const string_t &value, const LogicalTypeId &type) {
+	switch (type) {
+	case LogicalTypeId::BLOB:
+		return value.GetSize();
+	case LogicalTypeId::VARCHAR:
+		return LengthFun::Length<string_t, int64_t>(value);
+	default:
+		throw NotImplementedException(LogicalTypeIdToString(type));
+	}
```

Comment:
Is this (and the exception below) a valid path? If so, let's add a test.
Otherwise, let's turn them into internal exceptions.